testing/scenario_app/android/app/src/androidTest/java/dev/flutter/scenarios/EngineLaunchE2ETest.java

```diff
@@ -34,6 +36,12 @@ public void smokeTestEngineLaunch() throws Throwable {
     UiThreadStatement.runOnUiThread(() -> engine.set(new FlutterEngine(applicationContext)));
     CompletableFuture<Boolean> statusReceived = new CompletableFuture<>();
 
+    // Resolve locale to `en_US`.
+    // This is required, so `window.locale` in populated in dart.
+    UiThreadStatement.runOnUiThread(
```

Comment:
I'd just remove that check in main.dart if https://github.com/flutter/engine/pull/17473 is just for iOS. 

Our mock framework is a mock rather than a golden. We should reproduce framework logic as long as it's useful in testing the engine but it doesn't seem to be the case here.

Reply:
but that wouldn't test that `window.locale` is set on iOS. Is there any case where `window.locale` can be `null`, and it's expected behavior?